lib/services/files-api-service.js

```diff
@@ -0,0 +1,129 @@
+    new di.Inject(
+      'Constants',
+      'Promise',
+      'Services.Configuration',
+      'Services.Waterline'
+    )
+);
+
+
+function filesApiServiceFactory(
+    constants,
+    Promise,
+    configuration,
+    waterline
+) {
+    var walker;
+    var walk = Promise.promisifyAll(walking);
+    var skuPath = constants.HttpStatic.onhttpSkupack;
+    var staticPath = constants.HttpStatic.onhttpStaticHttp;
+    Promise.promisifyAll(fs);
+
+    function FilesApiService() {
+    }
+
+
+    FilesApiService.prototype.pairSkupackIds = function() {
+        return fs.readdirAsync(skuPath).then(function(list) {
+            return Promise.filter(list, function(item) {
+                return (fs.lstatAsync(path.join(skuPath, item)))
+                .then(function(stat) {
+                    return stat.isDirectory();
+                }); 
+            }).then(function(dirList) {
+                return Promise.reduce(dirList, function(result, item) {
+                    return waterline.skus.findOne({id : item})
+                    .then(function(sku) {
+                        result[item] = sku.name;
+                        return result;
+                    });
+                },{});
+            });
+        });
+    };
+
+
+    FilesApiService.prototype.walkDirectory = function(dir) {
+        return new Promise(function(resolve, reject) {
+            var walkResults = [];
+            walker = walk.walk(dir, {followLinks: false});
+
+            walker.on("file", function(root, fileStats, next) {
+                Promise.resolve().then(function() {
+                    var walkObj = {};
+                    if (root.indexOf(staticPath) > -1){
+                        root = root.replace(staticPath, '');
+                        walkObj.uri = path.join(root, fileStats.name);
+                        walkResults.push(walkObj);
+                    } else if ((root.indexOf(skuPath) > -1) && 
+                      (root.indexOf("static/common") > -1)){
+                        var filePath = path.join(root, fileStats.name).split("/");
+                        var skuId = filePath[1];
+                        var skuUri = filePath.slice(2).join("/");
+                        walkObj.uri = skuUri;
+                        walkObj.sku = skuId;
+                        walkResults.push(walkObj);
+                    }
+                }).then(function() {
+                    next();
+                }).catch(function(e) {
+                    reject(e);
+                });
+            });
+
+            walker.on("end", function() {
+                resolve(walkResults);
+            });
+        });
+    };
+
+
+    FilesApiService.prototype.getAllStaticFiles = function() {
+
+        var self = this;
+        var directories = [staticPath,
+                           configuration.get('httpStaticRoot', '/opt/monorail/static/http')];
+
+        return fs.lstatAsync(skuPath).then(function(stat) {
+            stat.isDirectory();
```

Comment:
This check is here to see if the path in `skuPath` exists. If it does not then an `ENOENT` error is thrown. If it does exists then it gets pushed to the directories array.

My concern here is to look within skupath when it exists. If anyone has a problem with this approach and a suggested alternative then please let me know.